kong/db/declarative/init.lua

```diff
@@ -395,34 +395,32 @@ local function cache_entities(entities)
 end
 
 
-function declarative.load_into_cache(entities, send_events)
+function declarative.load_into_cache_with_events(entities)
+
+  -- ensure any previous update finished (we're flipped to the latest page)
+  local _, err = kong.worker_events.poll()
+  if err then
+    return nil, err
+  end
+
+  post_upstream_crud_delete_events()
 
-  local ok, err = cache_entities(entities)
+  local ok
+  ok, err = declarative.load_into_cache(entities, SHADOW)
   if ok then
-    kong.cache:flip()
+    kong.worker_events.post("declarative", "flip_config", true)
+    kong.worker_events.poll()
```

Comment:
Shouldn't `kong.worker_events.poll()` be checked for errors like it's done before? If not, I suggest adding a comment explaining that errors are suppressed on purpose.

Reply:
@Tieske is it guaranteed as per the contract of the worker-events API that the worker who is sending the event will execute the event callback _immediately_ (i.e., before `post` returns)? That way I would not have to `poll` here (the remainder of this function requires the operation triggered by this post to have been executed).

Reply:
doing a 'post' implicitly does a 'poll' (it returns the results from that `poll` call). This can return any of the following values:

- `true`: the event has been posted and handlers executed (queue is empty)
- `false`: the event was queued, but recursively from an event-handler, so hasn't been executed. Will be done after the current handler returns.
- `nil+error`: something went wrong

So the `poll` call here is useless, since the previous call to `post` already called `poll`.

The caveat is that the events will only be executed if the current code does NOT run itself within an event-handler.

Reply:
(the contract here is that every worker executes the events in the exact same order, guaranteed)

Reply:
ok, "post implies poll" is the bit I needed — and I just realized it is in the docs, d'oh. Sorry about the noise! @kikito, I will remove this line and add error checking in `:post`. Thanks for the catch!